client/rpcproxy/rpcproxy_test.go

```diff
@@ -612,207 +587,83 @@ func TestRPCProxyInternal_New(t *testing.T) {
 	if p.shutdownCh == nil {
 		t.Fatalf("bad")
 	}
-}
 
-// func (p *RPCProxy) reconcileServerList(l *serverList) bool {
-func TestRPCProxyInternal_reconcileServerList(t *testing.T) {
-	tests := []int{0, 1, 2, 3, 4, 5, 10, 100}
-	for _, n := range tests {
-		ok, err := test_reconcileServerList(n)
-		if !ok {
-			t.Errorf("Expected %d to pass: %v", n, err)
+	for _, l := range []*serverList{p.activatedList, p.primaryServers, p.backupServers} {
+		if l == nil {
+			t.Fatalf("serverList nil")
 		}
-	}
-}
 
-func test_reconcileServerList(maxServers int) (bool, error) {
-	// Build a server list, reconcile, verify the missing servers are
-	// missing, the added have been added, and the original server is
-	// present.
-	const failPct = 0.5
-	p := testRPCProxyFailProb(failPct)
-
-	var failedServers, healthyServers []*ServerEndpoint
-	for i := 0; i < maxServers; i++ {
-		nodeName := fmt.Sprintf("s%02d", i)
-
-		node := &ServerEndpoint{Name: nodeName}
-		// Add 66% of servers to RPCProxy
-		if rand.Float64() > 0.33 {
-			p.activateEndpoint(node)
-
-			// Of healthy servers, (ab)use connPoolPinger to
-			// failPct of the servers for the reconcile.  This
-			// allows for the selected server to no longer be
-			// healthy for the reconcile below.
-			if ok, _ := p.connPoolPinger.PingNomadServer(p.configInfo.Region(), p.configInfo.RPCMajorVersion(), node); ok {
-				// Will still be present
-				healthyServers = append(healthyServers, node)
-			} else {
-				// Will be missing
-				failedServers = append(failedServers, node)
-			}
-		} else {
-			// Will be added from the call to reconcile
-			healthyServers = append(healthyServers, node)
+		if len(l.L) != 0 {
+			t.Fatalf("serverList.servers length not zero")
 		}
 	}
+}
 
-	// Randomize RPCProxy's server list
-	p.RebalanceServers()
-	selectedServer := p.FindServer()
-
-	var selectedServerFailed bool
-	for _, s := range failedServers {
-		if selectedServer.Key().Equal(s.Key()) {
-			selectedServerFailed = true
-			break
-		}
-	}
+// TestRPCProxy_Race is meant to be run with -race to find races in rpcproxy
```

Comment:
TODO: Remove and replace with a test of concurrent FindServer and Rebalance calls.